crates/turbopack-core/src/package_json.rs

```diff
@@ -0,0 +1,79 @@
+use std::{fmt::Write, ops::Deref};
+
+use anyhow::Result;
+use serde_json::Value as JsonValue;
+use turbo_tasks::{debug::ValueDebugFormat, primitives::StringVc, trace::TraceRawVcs};
+use turbo_tasks_fs::{FileContent, FileJsonContent, FileJsonContentReadRef, FileSystemPathVc};
+
+use super::issue::{Issue, IssueVc};
+
+#[derive(PartialEq, Eq, ValueDebugFormat, TraceRawVcs)]
+pub struct PackageJson(FileJsonContentReadRef);
+
+impl Deref for PackageJson {
+    type Target = JsonValue;
+    fn deref(&self) -> &Self::Target {
+        match &*self.0 {
+            FileJsonContent::Content(json) => json,
+            _ => unreachable!("PackageJson is guaranteed to hold Content"),
```

Comment:
Should we have:

```rust
pub enum FileJsonContent {
    Content(Arc<Value>),
    Unparseable(Box<UnparseableJson>),
    NotFound,
}
```

At the cost of an additional indirection, this would let us pass the internal `Value` around, avoiding this check entirely.

The alternative is some mapped `Arc` type.

In any case, I think this is the only place where this would be useful, so it's not particularly relevant.